src/resources/InternalBaseInterface.ts

```diff
@@ -0,0 +1,10 @@
+export interface UnrecommendedPaginated {

```

Comment:
Why do we have to put up with this?

The API should make the change on its side to match the Coveo standard.

Reply:
Agreed, I'll first confirm my theory that `perPage` is the "Coveo Standard :tm:".

@louis-bompart  todos: 
 - [ ] start discussion on the standard
 - [ ] check with Vault API owners for standardization
 - [ ] check with Catalog API owners for standardization. (bonus)

For the filename, I'm unsure, I think there could be value to have some 'internal' interfaces, so maybe keeping it just as InternalBaseInterface.ts should be OK.

However, I will :
 - [ ] rename `UnrecommendedPaginated` to `DeprecatedPaginated`

Reply:
Good checklist :+1:

Reply:
The formalization etc of the standard won't be done in a day, far from it 🤣 
So, while I started some discussion on the topic, I need this to go on.

Both Vault API and Catalog API are OK with the idea of standardization when the standards are in place.

So, for the time being, I'll still take the low hanging fruits, such as this PR 😄

Reply:
You already have my approval and very happy that standardization is coming here, good job!!